work/worker.go

```diff
@@ -471,7 +477,9 @@ func (self *worker) makeCurrent(parent *types.Block, header *types.Header) error
 	}
 	work := NewTask(self.config, types.MakeSigner(self.config, header.Number), stateDB, header)
 	if self.nodetype != common.CONSENSUSNODE {
+		// set the current block and header as pending block and header to support APIs requesting a pending block.
 		work.Block = parent
+		work.header = parent.Header()
```

Comment:
I didn't find any negative effect when setting the previous block header.

Reply:
@kjeom You mean that you can't find the current problem related to block header, right?

1. Currently, an incomplete block header is used for `makeCurrent` and the method set the header to `worker.current.header`.
https://github.com/klaytn/klaytn/blob/6833919f7de6acda298cd3c9ff779daf5cdbd93c/work/worker.go#L529-L543


3. `worker.snapshotBlock` consists of `worker.current.header` and others
https://github.com/klaytn/klaytn/blob/6833919f7de6acda298cd3c9ff779daf5cdbd93c/work/worker.go#L616-L620

4. `worker.snapshotBlock` can be returned as a pending block 
https://github.com/klaytn/klaytn/blob/6833919f7de6acda298cd3c9ff779daf5cdbd93c/work/worker.go#L220-L225

5. In the JSON marshaling logic, the pending block's header is used to calculate a block proposer address 
https://github.com/klaytn/klaytn/blob/6833919f7de6acda298cd3c9ff779daf5cdbd93c/api/api_ethereum.go#L1463-L1474

Therefore, an incomplete block header assigned to `worker.current.header` can cause panic. Since EN/PN doesn't finish the pending block generation process, an alternative header is needed for `worker.current.header`

Reply:
@aidan-kwon Oh, sorry for confusing. I just noted what I have finished for review.
Anyway thanks for explaining it in detail.